src/frontends/smt2/parser.cpp

```diff
@@ -0,0 +1,691 @@
+    void parse_push() { throw_parser_exception("push not yet supported"); }
+    void parse_reset() { throw_parser_exception("reset not yet supported"); }
+    void parse_reset_assertions() { throw_parser_exception("reset-assertions not yet supported"); }
+    void parse_set_info() {
+        lean_assert(curr_kind() == scanner::token_kind::SYMBOL);
+        lean_assert(curr_symbol() == g_token_set_info);
+        next();
+        parse_attributes("set-info"); // only one attribute is okay
+        check_curr_kind(scanner::token_kind::RIGHT_PAREN, "invalid set-info, ')' expected");
+        next();
+    }
+
+    void parse_set_logic() {
+        // Currently ignored (we deduce the logic automatically)
+        lean_assert(curr_kind() == scanner::token_kind::SYMBOL);
+        lean_assert(curr_symbol() == g_token_set_logic);
+        next();
+        check_curr_kind(scanner::token_kind::SYMBOL, "invalid set-logic command, symbol expected");
+        symbol sym = curr_symbol();
+        next();
+        check_curr_kind(scanner::token_kind::RIGHT_PAREN, "invalid set-option, ')' expected");
+        next();
+    }
+
+    void parse_set_option() {
+        lean_assert(curr_kind() == scanner::token_kind::SYMBOL);
+        lean_assert(curr_symbol() == g_token_set_option);
+        next();
+        check_curr_kind(scanner::token_kind::KEYWORD, "invalid set-option command, keyword expected");
+        symbol sym = curr_symbol();
+        next();
+        if (sym == g_token_use_locals) {
+            m_use_locals = true;
+        } else {
+            // TODO(dhs): just a warning?
+            throw_parser_exception(std::string("unsupported option: ") + sym);
+        }
+        check_curr_kind(scanner::token_kind::RIGHT_PAREN, "invalid set-option, ')' expected");
+        next();
+    }
+
+public:
+    // Constructor
+    parser(environment const & env, io_state & ios, std::istream & strm, char const * strm_name):
+        m_env(env), m_ios(ios), m_scanner(strm, strm_name) {}
+
+    // Entry point
+    bool operator()() {
+        scoped_expr_caching disable(false);
+        scoped_set_distinguishing_pp_options set(get_distinguishing_pp_options());
+
+        buffer<module_name> olean_files;
+        std::string base = dirname(get_stream_name().c_str());
+        optional<unsigned> k;
+
+        olean_files.push_back(module_name(k, name("init")));
+        olean_files.push_back(module_name(k, name({"algebra", "smt2"})));
+
+        unsigned num_threads = 0;
+        bool keep_imported_theorems = false;
+
+        m_env = import_modules(m_env, base, olean_files.size(), olean_files.data(), num_threads, keep_imported_theorems, m_ios);
+
+        bool ok = true;
+        try {
+            parse_commands();
+        } catch (exit_exception const & ex) {
+            return ok;
+        } catch (throwable const & ex) {
+            ok = false;
+            legacy_type_context tc(m_env, m_ios.get_options());
+            auto out = regular(m_env, m_ios, tc);
```

Comment:
Why are you using the legacy_type_context? We are going to delete it as soon as we have the new elaborator.